### PR TITLE
`listAdapterOf` DSL implementation

### LINKED ISSUE #25: [samples] : Create example with long/normal click listeners
I got this feedback from reddit:
```txt
Need example with long/normal click listeners + per-view-id click tracking 
(in case if item has "favorite" and "delete" buttons for example).
```
It will be great to show an example how to deal with click events.  With an example, `README` may also be improved.

### LINKED ISSUE #29: [kiel] : Make ListAdapter Composable
As an Android Developer, we would like to able to use Kiel with `ListAdapter `. 

syntax would be:
```kt
val adapter =listAdapter<...>{
diffUtil{...}
register{...}
}

```
In addition to this, it will be nice to have an example in the `samples` project.

### PR DESCRIPTION
**Pull request by ibrahimyilmaz**
## Proposed Changes/Issues Fixed

`listAdapterOf` DSL introduced.
example syntax:
```kt

val adapter =listAdapterOf<Any> {
            itemDiffUtil { RecyclerViewItemCallback() }
            register { 
                type {  }
                viewHolder {  }
                layoutResource {  }
                onViewHolderBound { vh, i, it ->  }
                onViewHolderCreated<> { vh, i, it ->  }
                onViewHolderBoundWithPayload { vh, i, it, list ->  }<> { vh, i, it ->  }
            }
        }
```

Example:

## General

  - [x] Change/Fix added to `CHANGELOG.md`
  - [x] Issue Link https://github.com/ibrahimyilmaz/kiel/issues/29
  - [ ] Current Sample was changed/ new sample is introcuded

## Testing
  - [x] `Unit-Test`s implemented
  - [x] `Monkey-Test` done
  - [x] Does your submission pass `./gradlew check`?

### CODE REVIEW

File: kiel/build.gradle [hunk 1]
```diff
@@ -63,4 +65,6 @@ dependencies {
     testImplementation "com.nhaarman.mockitokotlin2:mockito-kotlin:$versions.mockitoKotlinVersion"
     testImplementation "org.mockito:mockito-inline:$versions.mockitoInlineVersion"
     testImplementation "com.google.truth:truth:$versions.truthVersion"
+    testImplementation 'android.arch.core:core-testing:1.1.1'
```
Comment (ibrahimyilmaz) [inline kiel/build.gradle:68]: Move it to dependencies.gradle.

File: kiel/build.gradle [hunk 2]
```diff
@@ -63,4 +65,6 @@ dependencies {
...
+
```
Comment (ibrahimyilmaz) [inline kiel/build.gradle:69]: remove unnecessary line.

File: kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapter.kt [hunk 3]
```diff
@@ -4,8 +4,10 @@ import android.view.ViewGroup
 import androidx.annotation.MainThread
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.RecyclerView
+import me.ibrahimyilmaz.kiel.core.RecyclerViewHolder
+import me.ibrahimyilmaz.kiel.core.RecyclerViewHolderManager
 
-class RecyclerViewAdapter<T : Any, VH : RecyclerViewHolder<T>> private constructor(
+class RecyclerViewAdapter<T : Any, VH : RecyclerViewHolder<T>> constructor(
```
Comment (ibrahimyilmaz) [inline kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapter.kt:10]: put private back

File: kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapter.kt [hunk 4]
```diff
@@ -52,6 +54,7 @@ class RecyclerViewAdapter<T : Any, VH : RecyclerViewHolder<T>> private construct
     }
 
     companion object {
+
```
Comment (ibrahimyilmaz) [inline kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapter.kt:57]: remove unnecessary line

File: kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapterBuilder.kt [hunk 5]
```diff
@@ -1,63 +1,21 @@
 package me.ibrahimyilmaz.kiel.adapter
 
-import android.view.View
-import androidx.collection.SimpleArrayMap
-import kotlin.reflect.KFunction1
+import me.ibrahimyilmaz.kiel.core.*
```
Comment (ibrahimyilmaz) [inline kiel/src/main/java/me/ibrahimyilmaz/kiel/adapter/RecyclerViewAdapterBuilder.kt:3]: dont use * imports.

File: kiel/src/test/java/me/ibrahimyilmaz/kiel/listadapter/RecyclerViewListAdapterBuilderTest.kt [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Test
+    fun `Should build an instance of RecyclerViewListAdapter`() {
+        // GIVEN
+        recyclerViewListAdapterBuilder.register {
+            layoutResource { 1 }
+            viewHolder { ::TestRecyclerViewHolder }
+            type { Any::class.java }
+        }
+        // WHEN
```
Comment (ibrahimyilmaz) [inline kiel/src/test/java/me/ibrahimyilmaz/kiel/listadapter/RecyclerViewListAdapterBuilderTest.kt:30]: put empty line before `WHEN`

File: CHANGELOG.md [hunk 7]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 ### Development
+- listAdapterOf builder function introduced.
```
Comment (ibrahimyilmaz) [inline CHANGELOG.md:10]: introduced created

**Comment by jesusmartinoza** [comment]
Thanks @ibrahimyilmaz for this feature. How can I use it, listAdapterOf is not in master or develop?

**Comment by ibrahimyilmaz** [comment]
Hello Jesus,
listAdapterOf builder function deprecated in previous versions and replaced with `adapterOf`.

`adapterOf` in the background uses ListAdapter of RecyclerView Adapters.

So you can use it!
Sincerely
Ibrahim

**Pull request merged**